core/src/test/java/com/google/cloud/sql/core/ConnectorTest.java

```diff
@@ -80,6 +81,32 @@ public void create_throwsErrorForInvalidInstanceName() throws IOException {
     assertThat(ex).hasMessageThat().contains("Cloud SQL connection name is invalid");
   }
 
+  @Test
+  public void create_throwsErrorForInvalidTlsCommonNameMismatch()
+      throws IOException, InterruptedException {
+    // The server TLS certificate matches myProject:myRegion:myInstance
+    FakeSslServer sslServer = new FakeSslServer();
+    ConnectionConfig config =
+        new ConnectionConfig.Builder()
+            .withCloudSqlInstance("myProject:myRegion:otherInstance")
```

Comment:
nit: let's make this blatantly wrong, e.g., "wrongwrongwrong" or something similar that catches the eye.

Reply:
Fixed.